packages/e2e-tests/specs/block-grouping.test.js

```diff
@@ -0,0 +1,234 @@
+
+			const allBlocks = await getAllBlocks();
+
+			expect( allBlocks[ 0 ].name ).toBe( 'core/group' );
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'ungroups using keyboard shortcut', async () => {
+			// Create a Group
+			await insertBlocksOfMultipleTypes();
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+			await transformBlockTo( 'Group' );
+
+			// Ungroup (Primary, Shift, Alt)
+			await pressKeyWithModifier( 'secondary', 'g' );
+
+			const allBlocks = await getAllBlocks();
+
+			expect( allBlocks[ 0 ].name ).not.toBe( 'core/group' );
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+	} );
+
+	describe( 'Container Block availability', () => {
+		beforeAll( async () => {
+			// Disable the Group block
+			await page.evaluate( () => {
+				const { dispatch } = wp.data;
+				dispatch( 'core/edit-post' ).hideBlockTypes( [ 'core/group' ] );
+			} );
+		} );
+
+		beforeEach( async () => {
+			// Create a Group
+			await insertBlocksOfMultipleTypes();
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+		} );
+
+		afterAll( async () => {
+			// Re-enable the Group block
+			await page.evaluate( () => {
+				const { dispatch } = wp.data;
+				dispatch( 'core/edit-post' ).showBlockTypes( [ 'core/group' ] );
+			} );
+		} );
+
+		it( 'does not show group transform if container block is disabled', async () => {
+			const availableTransforms = await getAvailableBlockTransforms();
+
+			expect(
+				availableTransforms
+			).not.toEqual( expect.arrayContaining( [
+				'Group',
+			] ) );
+		} );
+
+		it( 'does not show group option in the options toolbar if container block is disabled ', async () => {
+			await clickBlockToolbarButton( 'More options' );
+
+			const blockOptionsDropdownHTML = await page.evaluate( () => document.querySelector( '.block-editor-block-settings-menu__content' ).innerHTML );
+
+			expect( blockOptionsDropdownHTML ).not.toContain( 'Group' );
+		} );
+	} );
+
+	describe( 'Preserving selected blocks attributes', () => {
+		it( 'preserves width alignment settings of selected blocks', async () => {
+			await insertBlock( 'Heading' );
+			await page.keyboard.type( 'Group Heading' );
+
+			// Full width image
+			await insertBlock( 'Image' );
+			await clickBlockToolbarButton( 'Full width' );
+
+			// Wide width image)
+			await insertBlock( 'Image' );
+			await clickBlockToolbarButton( 'Wide width' );
+
+			await insertBlock( 'Paragraph' );
+			await page.keyboard.type( 'Some paragraph' );
+
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await transformBlockTo( 'Group' );
+
+			const allBlocks = await getAllBlocks();
+
+			// We expect Group block align setting to match that
+			// of the widest of it's "child" innerBlocks
+			expect( allBlocks[ 0 ].attributes.align ).toBe( 'full' );
```

Comment:
This isn't needed as the snapshot contains the attribute.

Reply:
@talldan I agree that the snapshot somewhat covers this. However, in my experience snapshots are somewhat brittle and easy to ignore especially when you have a very specific piece of rendered output you want to assert against. 

In this case, I felt that adding an _explicit_ assertion here might be better. If I had to choose I'd probably opt to remove the snapshot.

What do you think?

Reply:
Definitely true that it can be hard to miss something missing in a snapshot. The assertion here is only testing that the store has the correct align attribute, so I think it's still valuable to retain the snapshot. Lets keep it as it is 👍